### ISSUE TITLE
'list -select' appears not to be operational

### ISSUE THREAD

**Issue by Ricky-Tigg** [openssl/openssl#19155]
_release:_ openssl-3.0.5
(output truncated):
```
$ man openssl-list
       -select name
           Only list algorithms that match this name.
```
Hello. With no known restriction to _name_ of option `-select`, it makes sense to select a name from a source made available, which is any produced by option *`-algorithms` and reported by `openssl help list`. The option appears not to be operational; **nothing printed**. Illustration (outputs truncated):
```
$ openssl list -digest-algorithms
Legacy:
  whirlpool
$ openssl list -select whirlpool
$ 
```

**Comment by levitte** [comment]
This isn't very well documented, but...  `-select` only works together with the diverse listing options, such as `-cipher-algorithms`, `-digest-algorithms`, etc.

I've noticed, though, that the implementation of `-select` is spotty, even with `-cipher-algorithms` et al.  In some cases, you get the whole list of algorithms instead of just the one selected.

**Comment by Ricky-Tigg** [comment]
I omitted to mention i had also executed `-select <name>` along with its relevant `-*-algorithms` in case it might have been meant to be operational that way. However `list -digest-algorithms -select whirlpool` behaved as `list -digest-algorithms`. Also the order of options mattered; illustration:
```
$ openssl list -select -digest-algorithms whirlpool
list: Use -help for summary.
```
For now, use of `-help` would not serve its purpose.

**Comment by Ricky-Tigg** [comment]
I omitted to mention that i had also executed `-select <name>` along with a relevant `-*-algorithms` in case it might have meant to be operational that way. However `list -digest-algorithms -select whirlpool` behaved as `list -digest-algorithms`.

**Comment by mattcaswell** [comment]
> I omitted to mention that i had also executed -select <name> along with a relevant -*-algorithms in case it might have meant to be operational that way. However list -digest-algorithms -select whirlpool behaved as list -digest-algorithms.

It only impacts provided algorithms. In the whirlpool case it does not exist in the default provider, so it never gets selected. It does exist in the "legacy" provider so this does print something:

````
$ openssl list -digest-algorithms -provider legacy  -select whirlpool
...
Provided:
  { 1.0.10118.3.0.55, whirlpool } @ legacy
````
It could definitely benefit from some improved documentation.

**Comment by Ricky-Tigg** [comment]
Correct me if i am wrong; that "`...`" is another illustration of non-queried complete list of algorithms. "`{ 1.0.10118.3.0.55, whirlpool } @ legacy`" alone would be the correct output.

**Comment by levitte** [comment]
> > Correct me if i am wrong; that "..." is another illustration of non-queried complete list of algorithms. "{ 1.0.10118.3.0.55, whirlpool } @ legacy" alone would be the correct output.
> 
> That's not an unreasonable way for things to work, but isn't the way "-select" currently operates. Whether that is a "bug" or not is debatable. Another approach is to add a parameter that suppresses legacy algorithms from the list output.

Debatable indeed.  There are places, specifically among the legacy functions, where the `-select` argument isn't looked at at all, while it is in others.  That's inconsistent.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of the 3.4 dev cycle barring further input
